common/src/main/java/org/mcaccess/minecraftaccess/features/NarrateHeldItem.java

```diff
@@ -3,56 +3,66 @@
-            // Narrate "empty slot" when the selected slot is empty
-            narrateIfHeldChanged("", 0, player.getInventory().getSelectedSlot(), NarrateHeldItem.EMPTY_SLOT_I18N);
-        }
+        ItemStack currentItemStack = player.getMainHandItem();
+        int selectedSlot = player.getInventory().getSelectedSlot();
+        String baseItemName = getItemName(currentItemStack);
+        int itemCount = currentItemStack.getCount();
 
-        if (!currentStackIsEmpty && player != null) {
-            // Narrate held item's name and count
-            narrateIfHeldChanged(currentStack.getHoverName().getString(), currentStack.getCount(), player.getInventory().getSelectedSlot(), NarrateHeldItem.HOTBAR_I18N);
-        }
-    }
+        String itemNameWithCount = currentItemStack.isStackable() ? itemCount + " " + baseItemName : baseItemName;
 
-    private void narrateIfHeldChanged(String itemName, int itemCount, int hotbarSlot, Function<String, String> i18n) {
-        boolean nameChanged = !previousItemName.equals(itemName);
-        boolean countChanged = previousItemCount != itemCount;
-        boolean slotChanged = previousHotbarSlot != hotbarSlot;
+        boolean nameChanged = !previousItemName.equals(baseItemName);
+        boolean countChanged = itemCount != previousItemCount;
+        boolean slotChanged = selectedSlot != previousSelectedSlot;
 
         if (nameChanged || slotChanged) {
-            String itemCountText = itemCount == 0 ? "" : NarrationUtils.narrateNumber(itemCount) + " ";
-            MainClass.narrate(i18n.apply(itemCountText + itemName), true);
+            MainClass.narrate(I18n.get("minecraft_access.other.selected", itemNameWithCount), true);
         } else if (countChanged && Config.getInstance().features.narrateHeldItemsCountWhenChanged) {
             MainClass.narrate(String.valueOf(itemCount), true);
         }
-        previousItemName = itemName;
+
+        previousItemName = baseItemName;
         previousItemCount = itemCount;
-        previousHotbarSlot = hotbarSlot;
+        previousSelectedSlot = selectedSlot;
+    }
+
+    private String getItemName(ItemStack itemStack) {
+        if (itemStack.isEmpty()) {
+            return I18n.get("minecraft_access.inventory_controls.empty_slot", "");
+        }
+
+        StringBuilder itemName = new StringBuilder();
+        itemName.append(itemStack.getHoverName().getString());
+
+        Optional<JukeboxPlayable> playableOpt = Optional.ofNullable(itemStack.get(DataComponents.JUKEBOX_PLAYABLE));
+        if (playableOpt.isPresent()) {
+            Optional<ResourceKey<JukeboxSong>> songKeyOpt = playableOpt.get().song().key();
+            if (songKeyOpt.isPresent()) {
+                String songPath = songKeyOpt.get().location().getPath();
+                String translatedDiscName = I18n.get("jukebox_song.minecraft." + songPath);
+                itemName.append(" ")
+                        .append(translatedDiscName);
+            }
+        }
```

Comment:
Adapted from InventoryControls:

https://github.com/minecraft-access/minecraft-access/blob/7dbe83180022c3efea7ef4fd2719d1b0cca90361/common/src/main/java/org/mcaccess/minecraftaccess/features/inventory_controls/InventoryControls.java#L501-L503

```suggestion
        Optional.ofNullable(itemStack.get(DataComponents.JUKEBOX_PLAYABLE))
                .flatMap(jukeboxPlayable -> jukeboxPlayable.song().key())
                .ifPresent(discNumber -> itemName.append(" ").append(I18n.get("jukebox_song.minecraft." + discNumber.location().getPath())));
```